### ISSUE TITLE
@no-named-arguments may violate implements/extends contract

### ISSUE THREAD

**Issue by klimick** [vimeo/psalm#5622]
In this situation usage of `no-named-arguments` violates contract of `Foo::foo`.
It requires `array<array-key, mixed>`. But `Bar::foo` requires more concrete type `list<mixed>`.
This snippet should trigger `MethodSignatureMismatch`. Actually no issues.

```php
<?php

interface Foo {
    public function foo(mixed ...$_args): void;
}

final class Bar implements Foo {
    /**
     * @no-named-arguments
     * !!! violates contract of Foo !!!
     */
    public function foo(mixed ...$_args): void
    {
    }
}

function getFoo(): Foo
{
    return new Bar();
}

$foo = getFoo();
$foo->foo(name: ''); // may cause runtime error
```
https://psalm.dev/r/c58ce00af0
